connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -271,15 +271,31 @@ message ExecutePlanRequest {
   // collate streaming responses from different queries within the dedicated session.
   string session_id = 1;
 
+  // (Optional)
+  // Provide an id for this request. It is needed if the request should be retryable.
+  optional string operation_id = 5;
+
   // (Required) User context
   //
   // user_context.user_id and session+id both identify a unique remote spark session on the
   // server side.
   UserContext user_context = 2;
 
-  // (Required) The logical plan to be executed / analyzed.
+  // The logical plan to be executed / analyzed.
+  // If the query has already been submitted, and this is a retry request, the plan is ignored.
   Plan plan = 3;
 
+  // (Optional)
+  // Specify if the request should be reattachable after e.g. a broken connection.
+  // If this is set to true, the client should call ReleaseRequest after it is done to let the
+  // server release resources; otherwise resources would be released only after a timeout.
+  optional bool reattachable = 6;
```

Comment:
In the spirit of simplicity, I wonder if we can "just do the right thing". Presumably if a query takes a while, the client wants an opId. If it completes in < x s, just do it inline/direct. Do we really need want an API for this?

Reply:
I added this to keep compatibility with the current semantics, which is push a complete stream of results.
Now I'm leaning that indeed, maybe having an option just like thrift direct results "return results inline if < x s and < y size" would be better.